### ISSUE TITLE
Prevent .osc directory from being pushed

### ISSUE THREAD

**Issue by dmach** [aplanas/openSUSE-git#71]
We shouldn't allow pushing changes with `.osc` directory because cloning such repo would only confuse osc with unexpected metadata.
Having a push hook guarding this would be nice.

Edit: osc sometimes even crashes when it bumps into unexpected `.osc` directory
It's important to scan all subdirs for the `.osc` directory and also all the new commits.

**Comment by JanZerebecki** [comment]
We can also make osc not become confused about this: By having osc creating .osc before cloning so git clone aborts the checkout (need to double check if this works) and by making osc check if .osc is in git and refuse to operate.

**Comment by dmach** [comment]
This is not about osc.
Users that use plain git can accidentally add and push `.osc` directory - and we don't want that.

**Comment by AdamMajer** [comment]
Agreed. This is the pre-commit hook requirement.

**Comment by AdamMajer** [comment]
If we have a pre-commit hook as 1st requirement, this is trivial to add there, so I'll add to that milestone too.

**Comment by JanZerebecki** [comment]
post-receive or similar server side git hooks are not documented for gitea, though I read that if you are admin you can configure them.
Is it enough if we reject PRs with this via CI and make the bot that merges things refuse if this was not successfully checked?

**Comment by AdamMajer** [comment]
well, gitea is run as a git-hook. Having this check as part of PR check makes sense. But pre-commit hook on client side would help at commit time (pre-push)

**Comment by josegomezr** [comment]
How about just using an action? make it exit !=0 if `.osc/` is detected (?)

**Comment by JanZerebecki** [comment]
It is not yet clear to me if we will have actions available, see: https://github.com/openSUSE/openSUSE-git/issues/78

**Comment by AdamMajer** [comment]
Actions are *after* upload. The requirement here is to prevent them from being pushed in the first place.

**Comment by JanZerebecki** [comment]
@dmach, I'm trying make osc error out if .osc is being version controlled by git in check_store_version() https://github.com/openSUSE/osc/blob/master/osc/obs_scm/store.py#L378 . But the osc git has quite a few .osc directories commited for test, so running those tests fail. How should we deal with those? Is it fine if I rename them dotosc and before they are used for tests prepare them to their real name?

If osc errors on version controlled .osc, then the precommit hook for running the appropriate osc services will fail, which will make git commit fail to commit an .osc if that hook is in use. And depending how we run the tests for Factory, those might use osc and thus fail merge request targeted for Factory.

**Comment by mmaslano** [comment]
@JanZerebecki this must be already done, right?

**Comment by JanZerebecki** [comment]
@mmaslano, I don't think this is done, unless @dmach implemented a solution in the mean time. I got no reply to my question from @dmach , and I'm hesitant to fully implement one approach without his preapproval as it is a lot of code to rewrite.

**Comment by mmaslano** [comment]
Daniel sadly agreed to fix this. Assigning to him.
